linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -224,6 +223,7 @@ where
         for blob in blobs {
             Self::check_blob_size(blob.content(), &policy)?;
         }
+        block.check_proposal_size(policy.maximum_block_proposal_size, blobs)?;
```

Comment:
Very small nit (for later, of course): I would actually swap the order we do these checks. We could have a lot of blobs that fit within the blob limit, but once put inside a proposal will not fit within the proposal limit. The other way around seems less likely, so we might fail earlier and save some decompressing time